### ISSUE TITLE
There seems to be no supported way to use Request withCredentials

### ISSUE THREAD

**Issue by jandppw** [SitePen/dstore#99]
dojo/request/xhr.__BaseOptions supports the withCredentials option, necessary for CORS behaviour.
There seems to be no supported way in dstore/Request to set this option, now any other option, like password, user, preventCache, sync, timeout, etc., apart from the headers and the query parameters.

			var response = request(requestUrl, {
				method: 'GET',
				headers: headers
			});

This makes Request and the specialisations of it unusable at least in CORS situations, unless I'm missing something.

**Comment by neonstalwart** [comment]
the resolution of #79 may provide a way for you to do it more directly through the store but, while the store itself might not expose what you need, it doesn't mean you can't do it - right now, i believe you can use a request provider to achieve what you want.

**Comment by kitsonk** [comment]
Yes, I agree that addressing #79 would provide a mechanism for this.  As @neonstalwart says, a request provider would be able to provide this mechanism for you, but I can see the benefit of making that more clean.
